crates/miden-lib/asm/miden/contracts/auth/basic.masm

```diff
@@ -12,6 +12,12 @@ const.AUTH_REQUEST=131087
 # The slot in this component's storage layout where the public key is stored.
 const.PUBLIC_KEY_SLOT=0
 
+const.SALT_PTR=0
+const.OUTPUT_NOTES_COMMITMENT_PTR=4
+const.INPUT_NOTES_COMMITMENT_PTR=8
+const.ACCOUNT_DELTA_COMMITMENT_PTR=12
+const.COMMITMENTS_END_PTR=16
```

Comment:
As discussed offline, instead of writing commitments to memory during execution, we could create a small helper procedure that could look something like:
```
#! Inputs:  [SALT, OUTPUT_NOTES_COMMITMENT, INPUT_NOTES_COMMITMENT, ACCOUNT_DELTA_COMMITMENT]
#! Outputs: [SALT, OUTPUT_NOTES_COMMITMENT, INPUT_NOTES_COMMITMENT, ACCOUNT_DELTA_COMMITMENT]
proc.adv_insert_hqword.16
    # TODO: write the values on the stack into the locals and then call adv.insert_mem
end
```
This procedure would then be called right after we build transaction summary (i.e., after line 52 below).

In the future, this procedure will be replaced with a single `adv.insert_hqword` instruction which would make it zero-cost (from the VM cycle standpoint). Let's create an issue for this in `miden-vm`.

Reply:
We still need to implement the procedure as a part of this PR, right?